### PR TITLE
build: Respect `native-*-tools-path` when building Wasm stdlib

### PR DESCRIPTION
**Pull request by kateinoigakukun**
This change allows the Wasm stdlib to be built using the prebuilt Swift toolchain instead of just built from source without building compiler tools. For example, I'm using the following build preset on my local.

```
skip-build-llvm
skip-build-swift
skip-build-cmark
skip-build-benchmarks
skip-early-swift-driver

build-wasm-stdlib

native-swift-tools-path=/path/to/swift-source/install/usr/bin
native-clang-tools-path=/path/to/swift-source/install/usr/bin
native-llvm-tools-path=/path/to/swift-source/install/usr/bin
```

### CODE REVIEW

**Comment by kateinoigakukun** [comment]
@swift-ci Please test WebAssembly

**Comment by kateinoigakukun** [comment]
@swift-ci smoke test

**Comment by kateinoigakukun** [comment]
@swift-ci Please test WebAssembly

**Comment by kateinoigakukun** [comment]
@swift-ci smoke test

**Comment by kateinoigakukun** [comment]
I'd like to merge this within a few days to unblock my threading support works on Wasm. This change only affects wasm build and unmaintained `stdlib_S_standalone_minimal_macho_x86_64`, so should be safe.

**Review by edymtt — COMMENTED** [review]
I would suggest to run the full macOS PR testing (since that will build `minimal-stdlib`, which uses the code path modified in `utils/swift_build_support/swift_build_support/cmake.py`), and the standalone stdlib PR testing (so we ensure that we are not regressing jobs using a prebuilt toolchain)

```
@swift-ci Please test stdlib with toolchain
@swift-ci Please test macOS platform
```

File: utils/swift_build_support/swift_build_support/cmake.py [hunk 1]
```diff
@@ -147,7 +147,7 @@ def common_options(self, product=None):
             define("CMAKE_CXX_COMPILER_LAUNCHER:PATH", args.cmake_cxx_launcher)
 
         if self.prefer_just_built_toolchain and product:
```
[the remark this answers is not in the document]
Comment (kateinoigakukun) [reply utils/swift_build_support/swift_build_support/cmake.py]: That makes sense to me 👍

**Comment by kateinoigakukun** [comment]
@swift-ci Please test stdlib with toolchain

**Review by edymtt — APPROVED** [review]

**Pull request merged**